packages/story-editor/src/components/canvas/pageSideMenu.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { ContextMenu } from '@web-stories-wp/design-system';
+import { __ } from '@web-stories-wp/i18n';
+import { useCallback } from '@web-stories-wp/react';
+import styled from 'styled-components';
+import { rgba } from 'polished';
+
+/**
+ * Internal dependencies
+ */
+import DirectionAware from '../directionAware';
+import { useLayout } from '../../app';
+import { useQuickActions } from '../../app/highlights';
+import { ZOOM_SETTING } from '../../constants';
+import PageMenu from './pagemenu/pageMenu';
+
+const MenusWrapper = styled.div`
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: space-between;
+  z-index: 9999;
+  pointer-events: auto;
+  min-height: 100%;
+  ${({ isZoomed, theme }) =>
+    isZoomed &&
+    `
+    min-height: initial;
+    background-color: ${rgba(theme.colors.standard.black, 0.45)};
+    border-radius: ${theme.borders.radius.small};
+    padding: 8px;
+    margin-left: -24px;
+  `}
+`;
+
+const Divider = styled.div`
+  width: 20px;
+  height: 1px;
+  margin: 13px auto 1px;
+  background-color: ${({ theme }) => theme.colors.divider.primary};
+`;
+
+function PageSideMenu() {
+  const { zoomSetting } = useLayout(({ state: { zoomSetting } }) => ({
+    zoomSetting,
+  }));
+  const quickActions = useQuickActions();
+
+  /**
+   * Stop the event from bubbling if the user clicks in between buttons.
+   *
+   * This prevents the selected element in the canvas from losing focus.
+   */
+  const handleMenuBackgroundClick = useCallback((ev) => {
+    ev.stopPropagation();
+  }, []);
+
+  const showQuickActions = Boolean(quickActions.length);
+
+  const isZoomed = zoomSetting !== ZOOM_SETTING.FIT;
```

Comment:
The layout sometimes looks weird in FIT because the available workspace area might just match the canvas dimensions leaving not enough space for the page menu:

![page-menu-fit-2](https://user-images.githubusercontent.com/637548/141402128-7bdc2264-5cd4-469d-80fe-7bf4ea3c322b.gif)

Observe how it jumps around a bit, sometimes not aligning both top and bottom to the canvas, sometimes partially under the design panel.

Reply:
Of course, this might be an acceptable edge case that we'll just file as a low-priority follow-up.

Reply:
I was wondering if this is something that should be part of the Page size reworking -- e.g. not allowing the Page grow so large that it doesn't allow the menu to fit. Thoughts?

Reply:
Hm, yeah that might be an idea - adding extra horizontal padding in FIT mode. Makes sense. Please make sure to capture this in #9642, otherwise I already forgot it 😃

Reply:
Done!